src/brain/index_selection.cpp

```diff
@@ -0,0 +1,449 @@
+        LOG_ERROR("Query is not bound");
+        assert(false);
+      }
+      IndexObjectPoolInsertHelper(tuple_child->GetBoundOid(), config);
+
+      break;
+    case ExpressionType::CONJUNCTION_AND:
+      PELOTON_FALLTHROUGH;
+    case ExpressionType::CONJUNCTION_OR:
+      left_child = where_expr->GetChild(0);
+      right_child = where_expr->GetChild(1);
+      IndexColsParseWhereHelper(left_child, config);
+      IndexColsParseWhereHelper(right_child, config);
+      break;
+    default:
+      LOG_ERROR("Index selection doesn't allow %s in where clause",
+                where_expr->GetInfo().c_str());
+      assert(false);
+  }
+  (void)config;
+}
+
+void IndexSelection::IndexColsParseGroupByHelper(
+    std::unique_ptr<parser::GroupByDescription> &group_expr,
+    IndexConfiguration &config) {
+  if ((group_expr == nullptr) || (group_expr->columns.size() == 0)) {
+    LOG_DEBUG("Group by expression not present");
+    return;
+  }
+  auto &columns = group_expr->columns;
+  for (auto it = columns.begin(); it != columns.end(); it++) {
+    assert((*it)->GetExpressionType() == ExpressionType::VALUE_TUPLE);
+    auto tuple_value = (expression::TupleValueExpression *)((*it).get());
+    IndexObjectPoolInsertHelper(tuple_value->GetBoundOid(), config);
+  }
+}
+
+void IndexSelection::IndexColsParseOrderByHelper(
+    std::unique_ptr<parser::OrderDescription> &order_expr,
+    IndexConfiguration &config) {
+  if ((order_expr == nullptr) || (order_expr->exprs.size() == 0)) {
+    LOG_DEBUG("Order by expression not present");
+    return;
+  }
+  auto &exprs = order_expr->exprs;
+  for (auto it = exprs.begin(); it != exprs.end(); it++) {
+    assert((*it)->GetExpressionType() == ExpressionType::VALUE_TUPLE);
+    auto tuple_value = (expression::TupleValueExpression *)((*it).get());
+    IndexObjectPoolInsertHelper(tuple_value->GetBoundOid(), config);
+  }
+}
+
+void IndexSelection::IndexObjectPoolInsertHelper(
+    const std::tuple<oid_t, oid_t, oid_t> tuple_oid,
+    IndexConfiguration &config) {
+  auto db_oid = std::get<0>(tuple_oid);
+  auto table_oid = std::get<1>(tuple_oid);
+  auto col_oid = std::get<2>(tuple_oid);
+
+  // Add the object to the pool.
+  IndexObject iobj(db_oid, table_oid, col_oid);
+  auto pool_index_obj = context_.pool_.GetIndexObject(iobj);
+  if (!pool_index_obj) {
+    pool_index_obj = context_.pool_.PutIndexObject(iobj);
+  }
+  config.AddIndexObject(pool_index_obj);
+}
+
+double IndexSelection::ComputeCost(IndexConfiguration &config, Workload &workload) {
+  double cost = 0.0;
+  auto queries = workload.GetQueries();
+  for (auto query : queries) {
+    std::pair<IndexConfiguration, parser::SQLStatement *> state = {config,
+                                                                   query};
+    if (context_.memo_.find(state) != context_.memo_.end()) {
+      cost += context_.memo_[state];
+    } else {
+      auto result =
+          WhatIfIndex::GetCostAndBestPlanTree(query, config, DEFAULT_DB_NAME);
```

Comment:
We are already starting to support multiple databases so we can't hard code a database name here.

Reply:
Done